src/uct/ib/configure.m4

```diff
@@ -197,12 +197,10 @@ AS_IF([test "x$with_ib" = "xyes"],
                       [], [], [[#include <infiniband/verbs.h>]])
 
        # Check ECE operation APIs are supported by rdma-core package
-       AC_CHECK_DECL(ibv_set_ece, [
-       AC_TRY_COMPILE([#include <infiniband/verbs.h>],
-                      [ibv_set_ece(NULL, NULL)],
-                      [AC_DEFINE([HAVE_VERBS_ECE], 1,
-                          [have upstream ECE APIs])])],
-                          [], [[#include <infiniband/verbs.h>]])
+       AC_CHECK_DECL(ibv_set_ece,
+                     [AC_DEFINE([HAVE_DECL_IBV_ECE], 1,
```

Comment:
HAVE_DECL__IBV_SET_ECE would be set automatically by AC_CHECK_DECL macro

Reply:
@yosefe Should we define ```#define HAVE_DECL_IBV_SET_ECE 0``` macro even ```ibv_set_ece``` isn't supported in the header file?
My original idea is 1) ```#define HAVE_DECL_IBV_ECE 1``` if ```ibv_set_ece``` is supported in the header file 2) do not define macro if ```ibv_set_ece``` is not supported instead of define some macro to be ```0```

Reply:
no; i mean that AC_CHECK_DECL already defines the macro. check online doc for AC_CHECK_DECL.

Reply:
yes

Reply:
@yosefe OK, do we need to define the macro **#define HAVE_DECL_IBV_SET_ECE 0** if ```ibv_set_ece``` API doesn't exist in the header file instead of not defining the macro at all? 
The current solution won't define any macro if ```ibv_set_ece``` API doesn't exist in the header file.

Reply:
no need to define it

Reply:
@yosefe If so, I'll keep the orginal the code here.

Reply:
still need to remove `AC_DEFINE([HAVE_DECL_IBV_ECE], 1,` because it would be defined by AC_CHECK_DECLS . the code can do `#ifdef HAVE_DECL__IBV_SET_ECE`

Reply:
ok

Reply:
@yosefe I'll follow your comment to use **AC_CHECK_DECLS** and ```#if(HAVE_DECL_IBV_SET_ECE)``` in code.

Reply:
Done